lib/mrsk/configuration/builder.rb

```diff
@@ -0,0 +1,113 @@
+class Mrsk::Configuration::Builder
+  def initialize(config:)
+    @options = config.raw_config.builder || {}
+    @image = config.image
+    @server = config.registry["server"]
+
+    valid?
+  end
+
+  def to_h
+    @options
+  end
+
+  def multiarch?
+    @options["multiarch"] != false
+  end
+
+  def local?
+    !!@options["local"]
+  end
+
+  def remote?
+    !!@options["remote"]
+  end
+
+  def cache?
+    !!@options["cache"]
+  end
+
+  def args
+    @options["args"] || {}
+  end
+
+  def secrets
+    @options["secrets"] || []
+  end
+
+  def dockerfile
+    @options["dockerfile"] || "Dockerfile"
+  end
+
+  def context
+    @options["context"] || "."
+  end
+
+  def local_arch
+    @options["local"]["arch"] if local?
+  end
+
+  def local_host
+    @options["local"]["host"] if local?
+  end
+
+  def remote_arch
+    @options["remote"]["arch"] if remote?
+  end
+
+  def remote_host
+    @options["remote"]["host"] if remote?
+  end
+
+  def cache_from
+    if cache?
+      case @options["cache"]["type"]
+      when 'gha'
+        "type=gha"
+      when 'registry'
+        [
+          "type=registry",
+          "ref=#{@server}/#{cache_image}"
+        ].compact.join(",")
+      end
+    end
+  end
+
+  def cache_to
+    if cache?
+      case @options["cache"]["type"]
+      when "gha"
+        [
+          "type=gha",
+          @options["cache"]&.fetch("options", nil),
+        ].compact.join(",")
+      when "registry"
+        [
+          "type=registry",
+          @options["cache"]&.fetch("options", nil),
+          "ref=#{@server}/#{cache_image}"
+        ].compact.join(",")
```

Comment:
Extract this as `cache_config_for_registry` or similar.

Reply:
Done.